runtime/jcl/common/thread.cpp

```diff
@@ -603,28 +612,131 @@ Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass unusedClass, jobje
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
 #if JAVA_SPEC_VERSION >= 20
+/**
+ * Frame walk iterator to find the most recent scoped value bindings on the stack.
+ *
+ * @param currentThread the current thread
+ * @param walkState the stack walk state
+ * @return J9_STACKWALK_KEEP_ITERATING or J9_STACKWALK_STOP_ITERATING
+ */
+static UDATA
+findScopedValueBindingsWalkFunction(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+	UDATA rc = J9_STACKWALK_KEEP_ITERATING;
+
+	if (NULL != walkState->userData1) {
+		J9Method *method = walkState->method;
+		J9Class *methodClass = J9_CLASS_FROM_METHOD(method);
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+		J9UTF8 *methodName = J9ROMMETHOD_NAME(romMethod);
+		J9ScopedValueBindingsWalkClasses *classes = (J9ScopedValueBindingsWalkClasses *)walkState->userData3;
+
+		if (((classes->threadClass == methodClass)
+				|| (classes->virtualThreadClass == methodClass)
+				|| (classes->scopedValueCarrierClass == methodClass))
+			&& J9UTF8_LITERAL_EQUALS(J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodName), "runWith")
+		) {
+			UDATA count = (UDATA)walkState->userData2;
+			rc = J9_STACKWALK_STOP_ITERATING;
+			/* runWith method should only have one instance of java.lang.ScopedValue$Snapshot. */
+			Assert_JCL_true((NULL != walkState->userData1) && (1 == count));
```

Comment:
Isn't the first part of this assertion guaranteed by line 627, allowing this to be reduced to this?
```c
			Assert_JCL_true(1 == count);
```

Reply:
Yes, I will open a PR to update the assertion.

Reply:
Opened https://github.com/eclipse-openj9/openj9/pull/18290 to incorporate the above feedback.